packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +266,374 @@ export default class VrackMoveDialogCtrl {
-            allServices.dedicatedServerInterface,
+            this.data.eligibleServices.dedicatedServerInterface,
             (serverInterface) => {
-              allServices.dedicatedServer.push(serverInterface);
+              dedicatedServer.push(serverInterface);
             },
           );
-
-          allServices.dedicatedServerInterface = [];
-        }
-
-        if (allServices?.ip?.length > 0) {
-          const ipv6List = allServices.ip.filter((ip) =>
-            this.constructor.isIPv6(ip),
-          );
-          angular.forEach(ipv6List, (ipv6) => {
-            this.loadIpRegion(ipv6.id);
-          });
+          // const dedicatedServer = this.data.eligibleServices.dedicatedServerInterface.map((serverInterface) => serverInterface.dedicatedServer);
+          this.data.eligibleServices.dedicatedServerInterface = [];
+          this.data.eligibleServices.dedicatedServer = dedicatedServer;
         }
+      });
+  }
 
-        return this.getAvailableServices(allServices);
+  // Get service info for each IP loadbalancing retrieved from API
+  updateIpLoadbalancingServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.ipLoadbalancing.map((serviceId) => {
+          return this.vrackService
+            .getIpLoadbalancing(serviceId)
+            .then((response) => {
+              const iplb = VrackMoveDialogCtrl.getIpLoadbalancingNiceName(
+                response.data,
+              );
+              return iplb;
+            });
+        }),
+      )
+      .then((ipLoadbalancing) => {
+        this.data.eligibleServices.ipLoadbalancing = ipLoadbalancing;
       });
   }
 
-  static isServiceAllowed(service) {
-    return (
-      !service.expired &&
-      [STATUS.ok, STATUS.delivered].includes(service.state) &&
-      service.status === STATUS.ok
+  updateIpServiceInfo() {
+    const ipUpdated = this.data.eligibleServices.ip.map((ip) => {
+      const formattedService = {
+        id: ip,
+        niceName: ip,
+        trueServiceType: TYPE_SERVICE.ip,
+      };
+      return formattedService;
+    });
+    this.data.eligibleServices.ip = ipUpdated;
+  }
+
+  updateIpv6ServiceInfo() {
+    const { ip } = this.data.eligibleServices;
+    const ipv6Updated = this.data.eligibleServices.ipv6.map((ipv6) => {
+      const formattedService = {
+        id: ipv6,
+        niceName: ipv6,
+        trueServiceType: TYPE_SERVICE.ip,
+      };
+      return formattedService;
+    });
+    this.data.eligibleServices.ip = ip.concat(ipv6Updated);
+  }
+
+  updateLegacyVrackServiceInfo() {
+    const legacyVrackUpdated = this.data.eligibleServices.legacyVrack.map(
+      (service) => {
+        const formattedService = {
+          id: service,
+          niceName: service,
+          trueServiceType: TYPE_SERVICE.legacyVrack,
+        };
+        return formattedService;
+      },
     );
+    this.data.eligibleServices.legacyVrack = legacyVrackUpdated;
+  }
+
+  updateEligibleServices(result) {
```

Comment:
This is quite a big function. it won't ease maintanability

Reply:
Yes we need to do many things.